Makefile

```diff
@@ -14,46 +14,135 @@
 # limitations under the License.
 #
 
+SHELL = /bin/bash
```

Comment:
I feel like this Makefile is getting overly complex.
An idea would be to split it out into 2 scripts:
One `configure` script, that takes the proper parameters and then feeds them into a `Makefile.in` file to create a `Makefile`.
the `configure` script could take "variant", "platform", etc. as parameters, do the proper argument checking, and then run a simple `make all`.
We could look at other ways to simplify it, but this would be one way to keep using a Makefile.

As of right now, this is quite hard to read & understand.
I don't know how much work that would take to refactor it though.

Reply:
I agree that this Makefile is pushing the boundaries of what seems reasonable. The idea of having a `configure` script is interesting. I'd have to play with it a bit to see how it looks.

I was also thinking about replacing `make` with `Gradle`, which I think is vastly better for writing build logic that falls outside the normal C/C++ model. I was a little reluctant to add a JRE as a build dependency, though. Also, while we use Gradle pretty widely at Delphix, it would be new in the Linux appliance build area.

Reply:
Yeah I'm not sure what are the requirements to run gradle. Maybe having a JRE dependency wouldn't be that bad if that's all it requires? As long as it's not picky on the JRE version, that we do not need to build the build scripts and we can easily add the JRE dependency to the Dockerfile, I'd not see much issues with that.